apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js

```diff
@@ -102,3 +102,28 @@ export async function publish(id) {
 
 	await eventClient.sendEvents(NSIP_EXAM_TIMETABLE, examTimetableItemsPayload, EventType.Publish);
 }
+
+/**
+ * Unpublishes an examination timetable. Does so by updating the examination timetable to be unpublished and
+ * sending the examination timetable items to the event queue to be unpublished on the front office.
+ *
+ * @param { String } id
+ * @returns {Promise<void>}
+ */
+export async function unPublish(id) {
+	const now = new Date();
+	const updatedExaminationTimetable = await examinationTimetableRepository.updateByCaseId(
+		+id,
+		// @ts-ignore
```

Comment:
FYI the way to fix this error is the `updateByCaseId` repo function should take an "input" type, rather than the full type (as for an update no fields are required).

See: https://github.com/Planning-Inspectorate/back-office/blob/main/docs/types.md#prisma-types

For this, you'd want:
` * @param {import('@prisma/client').Prisma.ExaminationTimetableUpdateInput} examinationTimetable`

Reply:
Nice one Ben, good idea.  Implemented, and for the other update exam timetable function too.